server/routes/contest.js

```diff
@@ -8,13 +8,17 @@ const {
   getContests,
   createSubmissionByContestId,
   getUserContests,
+  getContestsBySimpleSearch,
+  getContestsByAdvanceSearch,
 } = require('../controllers/contest');
 
 router.post('/', protect, createContest);
 router.route('/user-contests').get(protect, getUserContests);
 router.post('/:id', protect, updateContest);
 router.get('/:id', protect, getContest);
-router.get('/all/contests', protect, getContests);
+router.get('/all/contests', getContests);
 router.post('/:id/submission', protect, createSubmissionByContestId);
+router.post('/all/contests/simpleSearch', getContestsBySimpleSearch);
```

Comment:
The current solution is functional, good work with that. However, we have two new endpoints just for searching, maybe consider the usage of query parameters in the route (That way we don't need to create any new routes) :)

Reply:
changed to using one new endpoint, and add type null as parameter to determine which search should be used in controller